API/cpp/Inc/I2C.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "PinDefs.hpp"
+
+#include <vector>
+
+typedef enum I2CBus {
+    I2CBus1,
+    I2CBus4,
+} I2CBus;
+
+class I2C {
+public:
+    I2C(I2CBus i2cBus);
+    ~I2C();
+
+    void transmit(uint8_t address, uint8_t regAddr, uint8_t data);
```

Comment:
I changed this to pass address as an arg. SPI didn't like us reinitializing the pins multiple times, so this may be the right way to do it. Just should test it